include/stl2/view/drop.hpp

```diff
@@ -0,0 +1,228 @@
+// cmcstl2 - A concept-enabled C++ standard library
+//
+//  Copyright Casey Carter
+//  Copyright Eric Niebler
+//  Copyright Christopher Di Bella
+//
+//  Use, modification and distribution is subject to the
+//  Boost Software License, Version 1.0. (See accompanying
+//  file LICENSE_1_0.txt or copy at
+//  http://www.boost.org/LICENSE_1_0.txt)
+//
+// Project home: https://github.com/caseycarter/cmcstl2
+//
+#ifndef STL2_VIEW_DROP_HPP
+#define STL2_VIEW_DROP_HPP
+
+#include <stl2/detail/fwd.hpp>
+#include <stl2/detail/meta.hpp>
+#include <stl2/detail/semiregular_box.hpp>
+#include <stl2/detail/iterator/concepts.hpp>
+#include <stl2/detail/functional/invoke.hpp>
+#include <stl2/detail/range/access.hpp>
+#include <stl2/detail/range/concepts.hpp>
+#include <stl2/detail/view/view_closure.hpp>
+#include <stl2/view/all.hpp>
+#include <stl2/view/view_interface.hpp>
+
+STL2_OPEN_NAMESPACE {
+	namespace ext {
+		template <Range R>
+		requires
+			View<R>
+		struct drop_view;
+
+		template <Range R>
+		requires
+			View<R>
+		class __drop_view : public view_interface<drop_view<R>> {
+			using D = iter_difference_t<iterator_t<R>>;
+
+			template <bool Const>
+			struct __sentinel;
+		public:
+			__drop_view() = default;
+
+			constexpr __drop_view(R base, D count)
+				: base_(std::move(base)),
+				  count_(count)
+			{}
+
+			template <Range O>
+			requires
+				ext::ViewableRange<O> &&
+				_ConstructibleFromRange<R, O>
+			constexpr __drop_view(O&& o, D count)
+				: base_(view::all(std::forward<O>(o))),
+				  count_(count)
+			{}
+
+			constexpr R base() const
+			{
+				return base_;
+			}
+
+			constexpr auto begin()
+			{
+				return begin_impl(*this);
+			}
+
+			constexpr auto begin() const
+			requires
+				Range<R const>
+			{
+				return begin_impl(*this);
+			}
+
+			constexpr auto begin()
+			requires
+				SizedRange<R>
+			{
+				return begin_impl(*this);
+			}
+
+			constexpr auto begin() const
+			requires
+				SizedRange<R const>
+			{
+				return begin_impl(*this);
+			}
+
+			constexpr auto end()
+			{
+				return __sentinel<SimpleView<R>>(__stl2::end(base_));
+			}
+
+			constexpr auto end() const
+			requires
+				Range<R const>
+			{
+				return __sentinel<true>(__stl2::end(base_));
+			}
```

Comment:
Both of these `end`s are wrong.

Reply:
There's a "burning the candle at both `end`s" joke here somewhere.